oauth/src/OAuth/OAuth2.hs

```diff
@@ -38,29 +62,23 @@ type Env env
     , HTTP.Env env
     )
 
-newtype AuthCode = AuthCode Text
-  deriving newtype (JSON.ToJSON, JSON.FromJSON)
-  deriving stock (Eq, Generic, Show)
-
-data AuthenticationRequest = AuthenticationRequest
-  { clientId     :: Text
-  , clientSecret :: Text
-  , code         :: AuthCode
-  , grantType    :: Text
-  , nonce        :: Text
-  , redirectUri  :: Text
-  , state        :: [(Text, Text)]
+data TokenRequest = TokenRequest
+  { tokenEndpoint :: TokenEndpoint
+  , clientId      :: ClientId
+  , clientSecret  :: ClientSecret
+  , code          :: AuthCode
+  , grantType     :: GrantType
+  , nonce         :: Nonce
+  , redirectURI   :: RedirectURI
+  , state         :: State
```

Comment:
Question: Is the change in the State type a result of an interface change or the previous version was plain wrong.